pkg/process_manager_test.go

```diff
@@ -0,0 +1,108 @@
+package process_manager
+
+import (
+	"sync"
+	"testing"
```

Comment:
Just a tip since using the stdlib is fine, but feel free to include something like https://github.com/stretchr/testify if you'd like some testing helpers

Reply:
That is nicer, yeah. Added some commits for this.